### PR TITLE
1655 judicial mapping tests

### PR DESCRIPTION
**Pull request by ConorH994**
https://tools.hmcts.net/jira/browse/AM-1655

### CODE REVIEW

File: src/main/resources/validationrules/iac/iac-judicial-office-holder-mapping.drl [hunk 1]
```diff
@@ -189,6 +189,13 @@ when
   $jap: JudicialAccessProfile(appointment == "President of Tribunal", (serviceCode == null || serviceCode != "BFA1"),
    (endTime == null || endTime.compareTo(ZonedDateTime.now()) >= 0),validateAuthorisation(authorisations))
 then
+   Map<String,JsonNode> attribute = new HashMap<>();
+   attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("IA"));
+   attribute.put("region", JacksonUtils.convertObjectIntoJsonNode($jap.getRegionId()));
```
Comment (dharmendra8186) [inline src/main/resources/validationrules/iac/iac-judicial-office-holder-mapping.drl:194]: we don't need to add attribute in judicial office holder drool files.

File: src/main/resources/validationrules/iac/iac-judicial-org-role-mapping.drl [hunk 2]
```diff
@@ -36,8 +34,8 @@ then
    attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("IA"));
    attribute.put("region", JacksonUtils.convertObjectIntoJsonNode($joh.getRegionId()));
    attribute.put("baseLocation", JacksonUtils.convertObjectIntoJsonNode($joh.getBaseLocationId()));
-   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($joh.getPrimaryLocation()));
-   attribute.put("contractType", JacksonUtils.convertObjectIntoJsonNode("Salaried"));
+   attribute.put("contractType", JacksonUtils.convertObjectIntoJsonNode("SALARIED"));
```
Comment (dharmendra8186) [inline src/main/resources/validationrules/iac/iac-judicial-org-role-mapping.drl:37]: Why "SALARIED" is in Upper case?

File: src/main/resources/validationrules/iac/iac-judicial-org-role-mapping.drl [hunk 3]
```diff
@@ -36,8 +34,8 @@ then
    attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("IA"));
    attribute.put("region", JacksonUtils.convertObjectIntoJsonNode($joh.getRegionId()));
    attribute.put("baseLocation", JacksonUtils.convertObjectIntoJsonNode($joh.getBaseLocationId()));
-   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($joh.getPrimaryLocation()));
```
Comment (dharmendra8186) [inline src/main/resources/validationrules/iac/iac-judicial-org-role-mapping.drl:39]: Why have you removed primaryLocation?

File: src/main/resources/validationrules/iac/iac-judicial-org-role-mapping.drl [hunk 4]
```diff
@@ -74,11 +72,11 @@ then
...
```
Comment (dharmendra8186) [inline src/main/resources/validationrules/iac/iac-judicial-org-role-mapping.drl:77]: why remove primary location?

File: src/main/resources/validationrules/iac/iac-judicial-org-role-mapping.drl [hunk 5]
```diff
@@ -74,11 +72,11 @@ then
...
+   attribute.put("substantive", JacksonUtils.convertObjectIntoJsonNode("Y"));
    if($joh.getOffice().equals("IAC Tribunal Judge (Fee-Paid)")) {
-      attribute.put("contractType", JacksonUtils.convertObjectIntoJsonNode("Fee-Paid"));
+      attribute.put("contractType", JacksonUtils.convertObjectIntoJsonNode("FEE-PAID"));
```
Comment (dharmendra8186) [inline src/main/resources/validationrules/iac/iac-judicial-org-role-mapping.drl:77]: why Fee-paid in upper case?

File: src/main/resources/validationrules/iac/iac-judicial-org-role-mapping.drl [hunk 6]
```diff
@@ -113,10 +111,10 @@ then
    attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("IA"));
    attribute.put("region", JacksonUtils.convertObjectIntoJsonNode($joh.getRegionId()));
    attribute.put("baseLocation", JacksonUtils.convertObjectIntoJsonNode($joh.getBaseLocationId()));
-   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($joh.getPrimaryLocation()));
```
Comment (dharmendra8186) [inline src/main/resources/validationrules/iac/iac-judicial-org-role-mapping.drl:116]: why remove primary location?

File: src/main/resources/validationrules/iac/iac-judicial-org-role-mapping.drl [hunk 7]
```diff
@@ -224,8 +222,12 @@ then
...
-   attribute.put("contractType", JacksonUtils.convertObjectIntoJsonNode("Salaried"));
+   attribute.put("substantive", JacksonUtils.convertObjectIntoJsonNode("Y"));
+   if($joh.getOffice().equals("IAC Tribunal Judge (Fee-Paid)")) {
```
Comment (dharmendra8186) [inline src/main/resources/validationrules/iac/iac-judicial-org-role-mapping.drl:226]: Please remove  if condition to check contract type as judge role is  only applicable for Tribunal Judge(Salaried)
Just keep this attribute.put("contractType", JacksonUtils.convertObjectIntoJsonNode("Salaried"));

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialBase.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+import org.kie.api.command.Command;
+import org.kie.api.runtime.ExecutionResults;
+import org.kie.api.runtime.KieContainer;
+import org.kie.api.runtime.StatelessKieSession;
+import org.kie.internal.command.CommandFactory;
+import uk.gov.hmcts.reform.orgrolemapping.domain.model.FeatureFlag;
+import uk.gov.hmcts.reform.orgrolemapping.domain.model.JudicialAccessProfile;
+import uk.gov.hmcts.reform.orgrolemapping.domain.model.JudicialOfficeHolder;
+import uk.gov.hmcts.reform.orgrolemapping.helper.TestDataBuilder;
+
```
Comment (dharmendra8186) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialBase.java:15]: Please remove  this class as we have already DroolBase.Java.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialMappingTest.java [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+    void shouldReturnImmigrationJudgeRoles() {
+
+        judicialOfficeHolders.forEach(judicialOfficeHolder -> {
+            judicialOfficeHolder.setOffice("IAC Designated Immigration Judge");
+        });
+
+        //Execute Kie session
+        buildExecuteKieSession(getFeatureFlags("iac_1_1", true), false);
+
+        //Extract all created role assignments using the query defined in the rules.
```
Comment (dharmendra8186) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialMappingTest.java:135]: have you cover all scenarios based on excel sheet?

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/TestDataBuilder.java [hunk 10]
```diff
@@ -343,8 +344,14 @@ public static RefreshJobEntity buildRefreshJobEntity() {
...
     public static JudicialAccessProfile buildJudicialAccessProfile() {
-        return JudicialAccessProfile.builder().userId(id_1).roleId("8")
-                .baseLocationId("south-east")
+        return JudicialAccessProfile.builder()
+                .userId(id_1)
+                .roleId("84")
+                .contractTypeId("5")
+                .beginTime(ZonedDateTime.now(ZoneOffset.UTC).plusDays(1))
+                .endTime(ZonedDateTime.now(ZoneOffset.UTC).plusMonths(1))
+                .baseLocationId("1")
```
Comment (dharmendra8186) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/TestDataBuilder.java:353]: why we are adding one month in end date and one day in begin date?

**Review by dharmendra8186 — COMMENTED** [review]
Please don't merge this PR.As, we first need to fix review comments and take the latest from jrd_org_role_mapping_support branch to start writing junit.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/TestDataBuilder.java (hunk 10 above)
Comment (ConorH994) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/TestDataBuilder.java:353]: its just for testing purposes to keep them separate

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialMappingTest.java (hunk 9 above)
Comment (ConorH994) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialMappingTest.java:135]: can you send me this excel please?

**Pull request closed without merging**